src/SymfonyCache/TaggableStore.php

```diff
@@ -0,0 +1,493 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\SymfonyCache;
+
+use Psr\Cache\InvalidArgumentException;
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+use Symfony\Component\Cache\Adapter\TagAwareAdapter;
+use Symfony\Component\Cache\Adapter\TagAwareAdapterInterface;
+use Symfony\Component\Cache\PruneableInterface;
+use Symfony\Component\HttpFoundation\HeaderBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpCache\StoreInterface;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+use Symfony\Component\Lock\Factory;
+use Symfony\Component\Lock\LockInterface;
+use Symfony\Component\Lock\Store\FlockStore;
+use Symfony\Component\Lock\Store\SemaphoreStore;
+
+/**
+ * Implements a storage for Symfony's HttpCache that supports tagging.
+ *
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
+ */
+class TaggableStore implements StoreInterface
+{
+    const NON_VARYING_KEY = 'non-varying';
+    const COUNTER_KEY = 'write-operations-counter';
+
+    /**
+     * @var string
+     */
+    private $purgeTagsHeader;
+
+    /**
+     * @var int
+     */
+    private $pruneThreshold;
+
+    /**
+     * @var TagAwareAdapterInterface
+     */
+    private $cache;
+
+    /**
+     * @var Factory
+     */
+    private $lockFactory;
+
+    /**
+     * @var LockInterface[]
+     */
+    private $locks = [];
+
+    /**
+     * Constructor.
+     *
+     * @param string $cacheDir        The cache directory
+     * @param int    $pruneThreshold  The number of write operations until orphan data gets pruned (default: 500)
+     * @param string $purgeTagsHeader The tags header name
```

Comment:
lets use an options resolver for the prune threshold and the header

Reply:
To be honest, I don't like the options resolver. It's way too verbose. Real arguments are self documenting and better to read.

Reply:
i disagree. the problem is that now if i want to change the header, i MUST specify some threshold. and if we add more options, the problem increases. imho optionsresolver is a very clean way to specify what is allowed, and includes validation like the threshold must be > 0 or the header must be a string without whitespace.